PDA cart swapping, eject hotkey and verb; toggle light action button

cacogen opened the pull request:
<!-- Write **BELOW** The Headers and **ABOVE** The comments else it may not be viewable. -->
<!-- You can view Contributing.MD for a detailed description of the pull request process. -->

## About The Pull Request

- If you click your PDA with a cartridge and it already has one, you'll automatically eject it before putting the new one in 
- You can also eject the cartridge without opening the UI by using ctrl+shift+click or the Remove Cartridge verb
- PDAs now have a toggle light action button, like other light sources
- PDA UI correctly updates when ejecting or inserting a cartridge or ejecting an ID

<!-- Describe The Pull Request. Please be sure every change is documented or this can delay review and even discourage maintainers from merging your PR! -->

## Why It's Good For The Game

This has bugged me for years, so I'm assuming it bugs other people too.
<!-- Please add a short description of why you think these changes would benefit the game. If you can't justify it in words, it might not be worth adding. -->

## Changelog
:cl: cacogen
add: Ctrl+shift+click PDA to eject cartridge
add: Clicking PDA with cartridge swaps out current cartridge
add: Remove Cartridge verb for people using the right-click menu or the Object tab
add: PDA toggle light action button
fix: PDA UI now updates when ejecting or inserting cartridge or ejecting ID
/:cl:

<!-- Both :cl:'s are required for the changelog to work! You can put your name to the right of the first :cl: if you want to overwrite your GitHub username as author ingame. -->
<!-- You can use multiple of the same prefix (they're only used for the icon ingame) and delete the unneeded ones. Despite some of the tags, changelogs should generally represent how a player might be affected by the changes rather than a summary of the PR's contents. -->

cacogen committed:
PDA cart swapping, eject hotkey and verb

trollbreeder commented:
finally

Cyberboss commented on code/game/objects/items/devices/PDA/PDA.dm:
```diff
@@ -807,6 +815,18 @@ GLOBAL_LIST_EMPTY(PDAs)
 		to_chat(usr, "<span class='warning'>This PDA does not have a pen in it!</span>")
 
+/obj/item/pda/proc/eject_cart()
+	if(issilicon(usr) || !usr.canUseTopic(src, BE_CLOSE, FALSE, NO_TK))
```
Think telekenetics should apply here, no?

cacogen replied:
TK doesn't work with the ID or pen and letting it work with the cart makes it teleport into your hand. I'm too lazy to add a check to make it drop like IDs in computers.

Cyberboss commented on code/game/objects/items/devices/PDA/PDA.dm:
```diff
@@ -807,6 +815,18 @@ GLOBAL_LIST_EMPTY(PDAs)
 		to_chat(usr, "<span class='warning'>This PDA does not have a pen in it!</span>")
 
+/obj/item/pda/proc/eject_cart()
+	if(issilicon(usr) || !usr.canUseTopic(src, BE_CLOSE, FALSE, NO_TK))
```
This proc shouldn't use `usr` and should take a `mob/user` parameter.

cacogen committed:
Update PDA.dm and action.dm

- Toggle light action button
- Proper use of mob/user instead of usr when appropriate
- Fixes UI not updating when ejecting or inserting cartridge or ejecting ID
- Makes toggle_light item action work with PDAs

cacogen commented:
This should be good to go if you want to review it again @cyberboss
